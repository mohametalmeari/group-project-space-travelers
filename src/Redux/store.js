import { configureStore } from '@reduxjs/toolkit';
import missionReducer from './features/missions/missionSlice';

const store = configureStore({
  reducer: {
    mission: missionReducer,
  },
});

export default store;
