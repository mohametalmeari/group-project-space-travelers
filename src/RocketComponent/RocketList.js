import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRocket } from '../Redux/Rocket/rocketSlice';
import RocketItem from './RocketItem';

const RocketList = () => {
  const { rockets, isLodaing, error } = useSelector((store) => store.rocket);
  const dispatch = useDispatch();
  useEffect(() => {
    if (rockets.length === 0) {
      dispatch(fetchRocket());
    }
  }, [dispatch, rockets.length]);

  if (isLodaing) {
    return <p>is Loading...</p>;
  }
  if (error) {
    return <p>Error occurred while fetching rocket data.</p>;
  }
  return (
    <>
      <ul style={{ marginTop: '2%' }}>
        {rockets.map((rocketItems) => (
          <RocketItem key={rocketItems.id} rocketData={rocketItems} />
        )) }
      </ul>
    </>
  );
};

export default RocketList;
