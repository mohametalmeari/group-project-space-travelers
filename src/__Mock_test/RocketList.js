import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchRocket } from '../Redux/Rocket/rocketSlice';
import RocketItem from './RocketItem';

const RocketList = () => {
  const { rockets, isLodaing, error } = useSelector((store) => store.rocket);
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(fetchRocket());
  }, [dispatch]);
  if (isLodaing) {
    return <p>is Loading...</p>;
  }
  if (error) {
    return <p>Error occurred while fetching rocket data.</p>;
  }
  return (
    <>
      {rockets.map((rocketItems) => (
        <RocketItem key={rocketItems.id} rocketData={rocketItems} />
      )) }
    </>
  );
};

export default RocketList;
