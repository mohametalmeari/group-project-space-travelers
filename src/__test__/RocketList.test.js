import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../Redux/store';
import RocketList from '../__Mock_test/RocketList';

describe('Rocket List Component', () => {
  test('Test render of Rocket List', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <RocketList />
        </Provider>
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
