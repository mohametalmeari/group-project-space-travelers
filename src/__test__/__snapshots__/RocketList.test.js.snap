// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket List Component Test render of Rocket List 1`] = `<div />`;
